### PR TITLE
Master r1834

### PR DESCRIPTION
**Pull request by mbgrydeland**

### CODE REVIEW

File: bin/varnishd/cache/cache_session.c [hunk 1]
```diff
@@ -196,6 +196,7 @@ HTC_RxInit(struct http_conn *htc, struct ws *ws)
 	if (htc->pipeline_b != NULL) {
 		l = htc->pipeline_e - htc->pipeline_b;
 		assert(l > 0);
+		assert(l <= ws->r - htc->rxbuf_b);
```
Comment (Dridi) [inline bin/varnishd/cache/cache_session.c]: Considering this is an open issue, why not `xxxassert`?

**Comment by bsdphk** [comment]
OK

**Comment by mbgrydeland** [comment]
Has been merged

**Comment by hermunn** [comment]
Backport review: Already in 4.1.4 (c1fa3248590613 and a9e67666349b56).

**Pull request closed without merging**